### PR TITLE
Split host from prefix for key names

### PR DESCRIPTION
**Pull request by RAttab**
This enables the prometheus backend to store the host as part of the labels which makes life much easier. A default is also provided via the `gethostname` function. We will usually override this by using the `hostname -s` command.

This change requires the addition of a new field in the optics header to hold the host and the poller now passes the host as part of the poll struct to the various backends. Nothing too fancy here.

---
This change is [Reviewable](https://reviewable.io/reviews/rattab/optics/7)

### CODE REVIEW

**Comment by RAttab** [comment]
Note that I just noticed that the host and prefix will be inversed in the carbon output which is not a good thing. Currently fixing it.

**Comment by RAttab** [comment]
Alright, I fixed my fuck with prefix. Feel free to continue reviewing.

File: src/optics.c [hunk 1]
```diff
@@ -209,6 +216,30 @@ bool optics_set_prefix(struct optics *optics, const char *prefix)
...
 
+const char *optics_get_host(struct optics *optics)
+{
+    return optics->header->host;
+}
+
+bool optics_set_host(struct optics *optics, const char *host)
+{
+    if (strnlen(host, optics_name_max_len) == optics_name_max_len) {
+        optics_fail("host '%s' length is greater then max length '%d'",
```
Comment (haguenau) [inline src/optics.c]: Perhaps `>=`? Also s/then/than/.

File: src/optics.h [hunk 2]
```diff
@@ -192,7 +197,7 @@ bool optics_poll_normalize(
 
 struct optics_poller;
 
-struct optics_poller *optics_poller_alloc();
+struct optics_poller * optics_poller_alloc();
```
Comment (haguenau) [inline src/optics.h]: That's surprising.

Comment (RAttab) [inline src/optics.h]: Well, it brings it in lines with the rest of the code style but really has nothing to do with the rest of the PR. Just something I noticed and can't be bother to create a single commit  for.

File: src/optics.c (hunk 1 above)
Comment (RAttab) [inline src/optics.c]: According to the man pages, `strnlen` is not supposed to be able to return a value greater then `maxlen` so `==` should be sufficient.

That being said, I just noticed that `strncpy` is used in this function and in `optics_set_prefix`. That function has some rather surprising behavior (harmless in this case) so I'll replace by `strlcpy` which does the expected.

**Comment by RAttab** [comment]
Typo fixed and moved to `strlcpy` in both `optics_set_prefix` and `optics_set_host`. Let me know if there's anything else that needs fixing.

**Comment by haguenau** [comment]
I didn't see anything else. LGTM.

**Pull request merged**